### PR TITLE
New Media Picker: Updated the media picker keyboard sizing logic

### LINKED ISSUE #7756: Inline media picker incorrectly sized on iPad with external keyboard
On an iPad with a hardware keyboard connected (tested with smart keyboard cover), the media picker appears too small. I guess this is related to our recent [keyboard height fix](https://github.com/wordpress-mobile/WordPress-iOS/pull/7723), as the height seems to be matching the height of the quicktype / suggestions bar.

I'm not entirely sure if this is an iOS 11 issue or not, as I only have an iOS 11 iPad here to test on and I can't replicate in the simulator.

**To replicate:**

* Connect an iPad to a hardware keyboard
* Open Aztec
* Tap the + (media) button in the formatting toolbar

img_0233

I guess we either need to set a minimum height on the inline picker?

cc @bummytime 

##### Tested on iPad Pro 9.7", iOS 11 (beta 7), WPiOS `develop`

### PR DESCRIPTION
**Pull request by bummytime**
Fixes #7756

This PR updates the inline picker height logic that was introduced as part of #7723. A couple of changes were made:

1. The `currentKeyboardFrame`'s  coord system is converted from the window's thus providing more accurate values.
2. A computed var (`isExternalKeyboardConnected`) was added to determine if an external keyboard is currently connected.
3. The inline picker's height calculation now varies depending on if the external keyboard is present or not.

To test:

On the Sim and/or device, build and run the app.  Try a few of the following scenarios:

* (iPad with an external keyboard) Edit a post and bring up the inline picker — make sure the picker's height is appropriate. Try again, but disconnect the keyboard while the picker is visible.
* (iPad with soft keyboard) Edit a post and bring up the inline picker — make sure the picker's height is appropriate. Try again, but connect the keyboard while the picker is visible.
* Make sure iPhones still work as expected.
* Retest the fixes from #7710...make sure they all still work.

Needs review: @frosty (Thank you!!!!!1 🎉 🙇 )

### CODE REVIEW

**Comment by bummytime** [comment]
One small note: chatted offline with @frosty and the issue described above is on iOS11 Beta 8. iOS 10 seems to work just fine.

**Comment by frosty** [comment]
Sorry @bummytime for leaving out that important detail 🙈

**Comment by bummytime** [comment]
@frosty Ready for another 👀 ... should work on both 10 and 11 now.

File: WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift [hunk 1]
```diff
@@ -2389,11 +2381,41 @@ private extension AztecPostViewController {
...
+            // that is true and return the result.
+            if ((currentKeyboardFrame.origin.y + currentKeyboardFrame.height) > view.frame.height) {
+                keyboardHeight = (currentKeyboardFrame.maxY - view.frame.height)
+            } else {
+                keyboardHeight = (currentKeyboardFrame.height - Constants.toolbarHeight)
+            }
+        }
+
+        // Sanity check
+        if keyboardHeight < keyboardHeightDefault {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift]: Could do `keyboardHeight = max(keyboardHeight, keyboardHeightDefault)`?

**Pull request merged**